src/target/riscv/batch.h

```diff
@@ -14,6 +14,17 @@ enum riscv_scan_type {
 	RISCV_SCAN_TYPE_WRITE,
 };
 
+/* These types are used to specify how many JTAG RTI cycles to add after a
+ * scan.
+ */
+enum riscv_scan_delay_type {
+	RISCV_SCAN_DELAY_DM,
+	RISCV_SCAN_DELAY_ABSTRACT_COMMAND,
+	RISCV_SCAN_DELAY_SYSBUS_READ,
+	RISCV_SCAN_DELAY_SYSBUS_WRITE,
+	RISCV_SCAN_N_DELAY_TYPES,
```

Comment:
If you like:

```suggestion
	/* Learned delay needed for accessing debug module registers: */
	RISCV_SCAN_DELAY_BASE,
	/* Learned delay for execution of an abstract command: */
	RISCV_SCAN_DELAY_ABSTRACT_COMMAND,
	/* Learned delay for System Bus read operation: */
	RISCV_SCAN_DELAY_SYSBUS_READ,
	/* Learned delay for System Bus write operation: */
	RISCV_SCAN_DELAY_SYSBUS_WRITE,
	/* Number of enum items: */
	RISCV_SCAN_N_DELAY_TYPES,
```

Furthermore, the constants could be shortened this way: `RISCV_SCAN_DELAY_*` --> `RISCV_DELAY_*`, if you like.

```suggestion
	/* Learned delay needed for accessing debug module registers: */
	RISCV_DELAY_BASE,
	/* Learned delay for execution of an abstract command: */
	RISCV_DELAY_ABSTRACT_COMMAND,
	/* Learned delay for System Bus read operation: */
	RISCV_DELAY_SYSBUS_READ,
	/* Learned delay for System Bus write operation: */
	RISCV_DELAY_SYSBUS_WRITE,
	/* Number of enum items: */
	RISCV_N_DELAY_TYPES,
```

Reply:
Addressed. Though I have dropped "Learned" from the comments, since IMHO it's too specific. It is the delays to use. They are not necessarily learned -- they could be obtained through other means in the future.